backend/src/atlas_assistant/settings.py

```diff
@@ -21,26 +35,68 @@ class Settings(BaseSettings):
     oidc_url: str | None = None
     oauth_client_id: str | None = None
 
-    model_config = SettingsConfigDict(env_file=".env", extra="forbid")  # pyright: ignore[reportUnannotatedClassAttribute]
+    model_config = SettingsConfigDict(  # pyright: ignore[reportUnannotatedClassAttribute]
+        env_file=".env", extra="forbid", env_nested_delimiter="__"
+    )
 
-    def get_model(self) -> ChatMistralAI:
+    def get_model(self) -> BaseChatModel:
         """Returns the chat model as identified by these settings."""
-        return ChatMistralAI(
-            model_name=f"mistral-{self.chat_model_size}-latest",
-            api_key=self.mistral_api_key,
-            temperature=self.chat_model_temperature,
-        )
+        if isinstance(self.chat_model, MistralConfig):
+            return ChatMistralAI(
+                model_name=f"mistral-{self.chat_model.size}-latest",
```

Comment:
Is there any chance we may want to "revert" to a previous version of the model (ie, if the _next_ latest version becomes more error-prone than the current version). If so, would it be worth setting that as an added `env`-based setting?